packages/react/src/components/controls/MediaDeviceSelect.tsx

```diff
@@ -16,19 +16,20 @@ export function useMediaDevices({ kind }: { kind: MediaDeviceKind }) {
 export interface UseMediaDeviceSelectProps {
   kind: MediaDeviceKind;
   room?: Room;
+  track?: LocalAudioTrack | LocalVideoTrack;
 }
 
 /** @public */
-export function useMediaDeviceSelect({ kind, room }: UseMediaDeviceSelectProps) {
+export function useMediaDeviceSelect({ kind, room, track }: UseMediaDeviceSelectProps) {
   const roomContext = useMaybeRoomContext();
   // List of all devices.
   const deviceObserver = React.useMemo(() => createMediaDeviceObserver(kind), [kind]);
```

Comment:
in `createMediaDeviceObserver` there still is a call to `getUserMedia` with just `audio:true, video: false` (or vice versa). This coul still resutl in duplicated/simultaneous calls to `getUserMedia`.
They should be skipeed if `usePreviewTracks` is used. (except id the js-client-sdk is able to properly skip those duplicated calls. also accounting for similar constraints liek `{video: true, audio:true}` `{video: {deviceId:"123"}, audio:{deviceId:234}}`

Reply:
that's true, however for the `DeviceMenu` component this is solved, by only rendering the `MediaDeviceSelect` once the track is available, see https://github.com/livekit/components-js/pull/537/files#diff-5a2258ada71877932aa61092f670ecc56b371aab617a35c00f7bf68cc54e6ab2R96

Reply:
once that track is enabled the `getUserMedia` call in room.getLocalDevices would only happen if the first call to `enumerateDevices` only returns dummy data. If however, permissions for camera and/or microphone have been granted already, it will skip the `getUserMedia` call altogether. Device Ids should not be impacting this at this point

Reply:
That makes sense. I am sceptical however that this is obvious to sdk users who dont use the DeviceMenu component. I could see how adding `props.disabled` until the tracks have triggered the `getUserMedia` permissions is missed and as a result the user could be faced with two prompts (if a browser that does not handle this gracefully is used)

I could see that adding an option to `useMediaDeviceSelect` and `createMediaDeviceObserver` like `requestPermissions` could make this more transparent.
```ts
 /**
   * @requestPermissions this will call getUserMedia if the permissions are not yet given to enumerate the devices 
   * with device labels see: https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/enumerateDevices
   * in some browsers multiple calls to getUserMedia result in multiple permission prompts. It is adviced to 
   * disable requestPermissions until getUserMedia is called once with the desired devices outside of this hook.
   */
```

Reply:
Agree that making this dependent on `disabled` isn't the most obvious of choices. 

I like the idea to make that available additionally! Thanks a lot for the documentation text, copied 90% of that over :)